src-tauri/src/secrets_store.rs

```diff
@@ -340,6 +340,61 @@ pub fn get_export_secret_keys_for_group(
     // }
 }
 
+/// Stores the NWC (Nostr Wallet Connect) URI for a specific public key in the secrets store.
+///
+/// # Arguments
+///
+/// * `pubkey` - The public key to associate the NWC URI with
+/// * `nwc_uri` - The NWC URI to store
+/// * `data_dir` - Path to the data directory
+///
+/// # Returns
+///
+/// * `Result<()>` - Ok(()) if successful, or an error if the operation fails
+pub fn store_nwc_uri(pubkey: &str, nwc_uri: &str, data_dir: &Path) -> Result<()> {
+    let mut secrets = read_secrets_file(data_dir).unwrap_or(json!({}));
+    let key = format!("nwc:{}", pubkey);
+    let obfuscated_uri = obfuscate(nwc_uri, data_dir);
+    secrets[key] = json!(obfuscated_uri);
+    write_secrets_file(data_dir, &secrets)?;
+    Ok(())
+}
+
+/// Retrieves the NWC URI for a specific public key from the secrets store.
+///
+/// # Arguments
+///
+/// * `pubkey` - The public key to get the NWC URI for
+/// * `data_dir` - Path to the data directory
+///
+/// # Returns
+///
+/// * `Result<String>` - The NWC URI if found, or an error if not found or operation fails
+pub fn get_nwc_uri(pubkey: &str, data_dir: &Path) -> Result<String> {
+    let secrets = read_secrets_file(data_dir)?;
+    let key = format!("nwc:{}", pubkey);
+    let obfuscated_uri = secrets[key].as_str().ok_or(SecretsStoreError::KeyNotFound)?;
+    deobfuscate(obfuscated_uri, data_dir)
+}
+
+/// Removes the NWC URI for a specific public key from the secrets store.
+///
+/// # Arguments
+///
+/// * `pubkey` - The public key to remove the NWC URI for
+/// * `data_dir` - Path to the data directory
+///
+/// # Returns
+///
+/// * `Result<()>` - Ok(()) if successful, or an error if the operation fails
+pub fn remove_nwc_uri(pubkey: &str, data_dir: &Path) -> Result<()> {
+    let mut secrets = read_secrets_file(data_dir)?;
+    let key = format!("nwc:{}", pubkey);
+    secrets.as_object_mut().map(|obj| obj.remove(&key));
+    write_secrets_file(data_dir, &secrets)?;
+    Ok(())
```

Comment:
Do we need a lock here to prevent race conditions upon modifying the secrets file?

Reply:
We might double check it when we arrive to WhiteNoise PR